src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = Object.assign({}, state);
+  const resultArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const key in action.extraData) {
+          stateClone[key] = action.extraData[key];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const item of action.keysToRemove) {
```

Comment:
You should use proper variable names. For example for `const item` you can use more informative name such as `const key` or something else.